activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb

```diff
@@ -34,51 +34,63 @@ def initialize(*)
 
         def enable_query_cache!
           @query_cache_enabled[connection_cache_key(current_thread)] = true
-          connection.enable_query_cache! if active_connection?
         end
 
         def disable_query_cache!
           @query_cache_enabled.delete connection_cache_key(current_thread)
-          connection.disable_query_cache! if active_connection?
+          connection.clear_query_cache if active_connection?
         end
 
-        def query_cache_enabled
+        def query_cache_enabled?
           @query_cache_enabled[connection_cache_key(current_thread)]
         end
+
+        alias :query_cache_enabled :query_cache_enabled?
+        deprecate :query_cache_enabled
       end
 
-      attr_reader :query_cache, :query_cache_enabled
+      attr_reader :query_cache
+
+      QC_ON  = Struct.new(:query_cache_enabled?).new(true)
+      QC_OFF = Struct.new(:query_cache_enabled?).new(false)
```

Comment:
☃️ Let's freeze these to make it clear they are constants

Reply:
`private_constant :QC_ON, :CQ_OFF` too 🔒